tests/src/Interop/StringMarshalling/LPSTR/LPSTRTest.cs

```diff
@@ -192,10 +192,19 @@ public static int Main(string[] args)
         }
 
         #region ReversePinvoke
-        DelMarshal_InOut d1 = new DelMarshal_InOut(Call_DelMarshal_InOut);
-        if (!PInvokeDef.RPinvoke_DelMarshal_InOut(d1, "ň"))
+        bool inOutByValueThrows = false;
+        try
         {
-            ReportFailure("Method RPinvoke_DelMarshal_InOut[Managed Side],Return value is false");
+            DelMarshal_InOut d1 = new DelMarshal_InOut(Call_DelMarshal_InOut);
```

Comment:
Can we use `Assert.Throws<>()`?

Reply:
I can re-write the tests to use the `Assert` infrastructure we have. Although I'd rather do that in a separate PR to keep this one focused on the string marshalling change.